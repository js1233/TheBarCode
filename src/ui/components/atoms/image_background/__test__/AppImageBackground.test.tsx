import { fireEvent, render } from "@testing-library/react-native";
import RightIcon from "assets/images/right.svg";
import React from "react";
import {
  AppImageBackground,
  CONTAINER_TYPES
} from "ui/components/atoms/image_background/AppImageBackground";

const rightSvgIcon = () => {
  return <RightIcon testID="icon" width={20} height={20} />;
};

test("snapshot testing", () => {
  const rendered = render(
    <AppImageBackground containerShape={CONTAINER_TYPES.CIRCLE} />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("renders icon correctly", () => {
  const { queryByTestId } = render(
    <AppImageBackground
      containerShape={CONTAINER_TYPES.CIRCLE}
      icon={rightSvgIcon}
    />
  );
  let icon = queryByTestId("icon");
  expect(icon).not.toBeNull();
});

it("should properly perform click event", () => {
  const onPress = jest.fn();
  const { getByTestId } = render(
    <AppImageBackground
      containerShape={CONTAINER_TYPES.CIRCLE}
      onPress={onPress}
      icon={rightSvgIcon}
    />
  );
  const imageButton = getByTestId("image-container");
  fireEvent.press(imageButton);
  expect(onPress).toBeCalledTimes(1);
});
