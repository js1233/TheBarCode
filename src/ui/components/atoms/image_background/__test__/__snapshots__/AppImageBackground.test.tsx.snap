// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "elevation": 3,
        "height": 40,
        "justifyContent": "center",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
        "width": 40,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "shadowColor": "#FFFFFF",
      },
      Object {
        "borderRadius": 50,
      },
      undefined,
    ]
  }
/>
`;
